cpp/util/SnowflakeCommon.cpp

```diff
@@ -119,5 +122,19 @@ CURLcode set_curl_proxy(CURL *curl, const char* proxy, const char* no_proxy)
   }
 }
 
+void STDCALL sf_disable_exit_on_memory_failure()
+{
+  // should be called by odbc driver snowflake_global_init
+  exit_on_memory_error = false;
+}
+
+void STDCALL sf_memory_error_handler()
+{
+  if (exit_on_memory_error)
+    exit(1);
+
+  throw std::runtime_error("fail to alloc memory");
```

Comment:
Would that be easily handled on ODBC side?
 `throw std::bad_alloc()` in existing implementation would be better than this one. Or maybe we need a new specific exception (could take SnowflakeTransferException as an example) so it could be easier for ODBC to catch and handle it?
 Please check existing implementation and make your changes inline with it.

Reply:
bad_alloc is fine. By the way, what do you men inlinez?

Reply:
I mean check the existing implementation first before making changes, try do the same/similar thing in the same/similar way, which would have lower risk to break anything.